Incremental location option

kathancox opened the pull request:
This PR covers a couple of different, related roadmap features

`RESTORE FROM LATEST` / `SHOW BACKUP LATEST` syntax 
Fixes [DOC-2623](https://cockroachlabs.atlassian.net/browse/DOC-2623) v22.1, v21.2, v21.1
- Added this change to the parameters table as a field (fast-follow to this PR will be to updated diagrams)
- Updated `RESTORE` syntax across the docs to _kind of_ have this syntax as the default in examples, with links to restoring a specific backup where it makes sense. 

`incremental_location` option 
Fixes [DOC-2087](https://cockroachlabs.atlassian.net/browse/DOC-2087) [DOC-2210](https://cockroachlabs.atlassian.net/browse/DOC-2210) [DOC-2615](https://cockroachlabs.atlassian.net/browse/DOC-2615) [DOC-2762](https://cockroachlabs.atlassian.net/browse/DOC-2762) v22.1 only
- Added this option to the `BACKUP`, `RESTORE`, `SHOW BACKUP` pages with examples & to option table
- Added a fuller explanation to the Take Full and Incremental Backups page 

`/incrementals` default directory
Fixes [DOC-2891](https://cockroachlabs.atlassian.net/browse/DOC-2891) v22.1 only
- Added to Take Full and Incremental Backups page 
- As well as to `BACKUP`, `RESTORE`

Deprecation notice for `BACKUP TO` and `RESTORE FROM` 
Fixes [DOC-2967](https://cockroachlabs.atlassian.net/browse/DOC-2967) v22.1, v21.2, v21.1
- Updated the language in the note.
-  This in 22.1 this is a warning

Deprecation notice for `SHOW BACKUP` 
Fixes [DOC-3115](https://cockroachlabs.atlassian.net/browse/DOC-3115) and [DOC-3114](https://cockroachlabs.atlassian.net/browse/DOC-3114)

Updated `SHOW BACKUP` syntax only on v22.1
Fixes [DOC-2623](https://cockroachlabs.atlassian.net/browse/DOC-2623)

Also added a section on Backup Collections to the Take full and Incremental Backups page. This explanation differs between v22.1 & v21.2/21.1.

**Note**: NOT included in this PR is `SHOW BACKUP` work with locality-aware backups that's to come in a separate PR and `SHOW BACKUP` diagram update will be a relatively fast follow PR.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/docs/13448)
<!-- Reviewable:end -->

kathancox commented:
Here are build links to some of the most-changed pages in v22.1
https://deploy-preview-13448--cockroachdb-docs.netlify.app/docs/dev/take-full-and-incremental-backups.html 
https://deploy-preview-13448--cockroachdb-docs.netlify.app/docs/dev/backup.html
https://deploy-preview-13448--cockroachdb-docs.netlify.app/docs/dev/restore.html

kathancox commented:
Hey @msbutler, @benbardin, @livlobo, I've added you as reviewers on this PR — if you think anyone else should review, please feel free to add. As I note above, upcoming changes to `SHOW BACKUP` syntax will be in a follow-up PR and updates to sql diagrams will also fast follow this PR.

benbardin commented on v21.1/backup.md:
```diff
@@ -58,7 +65,7 @@ To view the contents of an Enterprise backup created with the `BACKUP` statement
 -----------+-------------
 `targets` | Back up the listed [targets](#targets).
-`subdirectory` | The name of the specific subdirectory (e.g., `2021/03/23-213101.37`) where you want to add an [incremental backup](take-full-and-incremental-backups.html#incremental-backups). To view available subdirectories, use [`SHOW BACKUPS IN destination`](show-backup.html). If the `subdirectory` is not provided, a [full backup](take-full-and-incremental-backups.html#full-backups) will be c ...
+`subdirectory` | The name of the specific backup (e.g., `2021/03/23-213101.37`) in the collection to which you want to add an [incremental backup](take-full-and-incremental-backups.html#incremental-backups). To view available backup subdirectories, use [`SHOW BACKUPS IN destination`](show-backup.html). If the backup `subdirectory` is not provided, a [full backup](take-full-and-incremental-backups ...
```
@msbutler Not part of this diff strictly speaking, but do we allow arbitrary strings on purpose or as a side effect? Almost feels like we should block this, if the subsequent backup will just get lost.

benbardin commented on v22.1/take-full-and-incremental-backups.md:
```diff
@@ -14,10 +14,59 @@ There are two main types of backups:
+    |—— 09-155340.13/
+      |—— Full backup files
+[...]
+|—— incrementals
```
Yup! Nice!

benbardin reviewed (approved):
LGTM, but please also wait for Michael!

msbutler commented on v22.1/show-backup.md:
```diff
@@ -23,10 +23,11 @@ The `SHOW BACKUP` [statement](sql-statements.html) lists the contents of a backu
+`SHOW BACKUP location` | Show the details of the backup in the given [`location`](backup.html#backup-file-urls). [See the example](#show-a-backup).
+`SHOW BACKUP LATEST IN location` | Show the most recent backup added in the given [`location`](backup.html#backup-file-urls). [See the example](#show-the-most-recent-backup)
+`SHOW BACKUP SCHEMAS location` | Show the schema details of the backup in the given [`location`](backup.html#backup-file-urls). [See the example](#show-a-backup-with-schemas).
+`SHOW BACKUP subdirectory IN location` |  List the full and incremental backups that are stored in the given full backup's `subdirectory` within a [`location`](backup.html#backup-file-urls). [See the example](#show-details-for-scheduled-backups).
```
so, we should also add a deprecation warning for old style show backup

msbutler commented on v22.1/show-backup.md:
```diff
@@ -294,9 +328,9 @@ Or, use the `kms` option and the same KMS URI that was used to create the backup
+{% include_cached copy-clipboard.html %}
 ~~~ sql
-SHOW BACKUP '/2021/11/15-150703.21' IN 's3://test/backup-test?AWS_ACCESS_KEY_ID=[placeholder]&AWS_SECRET_ACCESS_KEY=[placeholder]' WITH debug_ids;
+SHOW BACKUP '/2021/11/15-150703.21' IN 's3://{bucket name}?AWS_ACCESS_KEY_ID=[placeholder]&AWS_SECRET_ACCESS_KEY=[placeholder]' WITH debug_ids;
```
nit: add FROM keyword

msbutler commented on v22.1/show-backup.md:
```diff
@@ -335,9 +369,9 @@ The response returned from `SHOW BACKUP ... WITH as_json` is a backup's internal
+{% include_cached copy-clipboard.html %}
 ~~~ sql
-SELECT jsonb_pretty(manifest->'entryCounts') AS f FROM [SHOW BACKUP '/2021/11/15-150703.21' IN 's3://test/backup-test?AWS_ACCESS_KEY_ID=[placeholder]&AWS_SECRET_ACCESS_KEY=[placeholder]' with as_json];
+SELECT jsonb_pretty(manifest->'entryCounts') AS f FROM [SHOW BACKUP '/2021/11/15-150703.21' IN 's3://{bucket name}?AWS_ACCESS_KEY_ID=[placeholder]&AWS_SECRET_ACCESS_KEY=[placeholder]' with as_json];
```
nit: add FROM keyword to SHOW BACKUP stmt

msbutler commented on v22.1/show-backup.md:
```diff
@@ -350,9 +384,9 @@ SELECT jsonb_pretty(manifest->'entryCounts') AS f FROM [SHOW BACKUP '/2021/11/15
+{% include_cached copy-clipboard.html %}
 ~~~ sql
-SELECT f->>'path' FROM (SELECT jsonb_array_elements(manifest->'files') AS f FROM [SHOW BACKUP '/2021/11/15-150703.21' IN 's3://test/backup-test?AWS_ACCESS_KEY_ID=[placeholder]&AWS_SECRET_ACCESS_KEY=[placeholder]' WITH as_json]);
+SELECT f->>'path' FROM (SELECT jsonb_array_elements(manifest->'files') AS f FROM [SHOW BACKUP '/2021/11/15-150703.21' IN 's3://{bucket name}?AWS_ACCESS_KEY_ID=[placeholder]&AWS_SECRET_ACCESS_KEY=[placeholder]' WITH as_json]);
```
nit: add FROM kewyord

msbutler commented on v22.1/take-and-restore-encrypted-backups.md:
```diff
@@ -161,6 +161,8 @@ For example, the encrypted backup created in the previous example can be restore
 (1 row)
 ~~~
 
+To restore from a specific backup, use [`RESTORE FROM {subdirectory} IN ...`](restore.html#restore-a-specific-backup).
```
nit: remove "specific"? a user may get the wrong idea that they can restore a single incremental backup or something.

msbutler commented on v22.1/restore.md:
```diff
@@ -64,6 +67,7 @@ You can control `RESTORE` behavior using any of the following in the `restore_op
 `encryption_passphrase`                                             | Passphrase used to create the [encrypted backup](take-and-restore-encrypted-backups.html) |  The passphrase used to decrypt the file(s) that were encrypted by the [`BACKUP`](take-and-restore-encrypted-backups.html) statement.
 `DETACHED`                                                          | N/A                                         |  When `RESTORE` runs with `DETACHED`, the job will execute asynchronously and the job ID will be returned immediately without waiting for the job to finish. Note that with `DETACHED` specified, further job information and the job completion status will not be returned. For more on t ...
 `debug_pause_on`                                                    | `"error" `                                    |  Use to have a `RESTORE` [job](show-jobs.html) self pause when it encounters an error. The `RESTORE` job can then be [resumed](resume-job.html) after the error has been fixed or [canceled](cancel-job.html) to rollback the job. <br><br>Example: `WITH debug_pause_on='error'`
+`incremental_location`<a name="incr-location"></a> | [`STRING`](string.html) | Restore an incremental backup from the alternate location the backup was originally taken with. <br><br>See [Restore incremental backups](#restore-from-incremental-backups) for more detail.
```
see my comment in the backup.md definition of this.

msbutler commented on v22.1/restore.md:
```diff
@@ -216,11 +227,11 @@ The examples in this section use the **default** `AUTH=specified` parameter. For
 ### View the backup subdirectories
 
-`BACKUP ... INTO` adds a backup to a collection within the backup destination. The path to the backup is created using a date-based naming scheme. To view the backup paths in a given destination, use [`SHOW BACKUPS`](show-backup.html):
+`BACKUP ... INTO` adds a backup to a [backup collection](take-full-and-incremental-backups.html#backup-collections) location. The path to a backup is created using a date-based naming scheme. To view the backup paths in a given collection location, use [`SHOW BACKUPS`](show-backup.html):
```
i don't think the second sentence is needed? also it's not always correct?

msbutler commented on v22.1/restore.md:
```diff
@@ -232,32 +243,41 @@ The examples in this section use the **default** `AUTH=specified` parameter. For
+When you want to [restore a specific backup](#restore-a-specific-backup), add the backup's subdirectory path (e.g., `/2021/12/21-142943.73`) to the `RESTORE` statement. For details on viewing the most recent backup, see [`SHOW BACKUP LATEST`](show-backup.html#show-the-most-recent-backup).
 
-Incremental backups will be appended to the full backup with `BACKUP ... INTO LATEST IN {destination}`. Your storage location will contain the incremental as a date-based subdirectory within the full backup.
+Incremental backups will be appended to the full backup with `BACKUP ... INTO LATEST IN {collection location}`. Your storage location will contain the incremental in the default `/incrementals` directory at the collection location's root. For more detail on collections, see [Backup Collections](take-full-and-incremental-backups.html#backup-collections).
```
I don't think it's necessary to include this info in Restore.md as they can look at SHOW BACKUP latest for more on how it displays incremental backups?

msbutler commented on v22.1/restore.md:
```diff
@@ -232,32 +243,41 @@ The examples in this section use the **default** `AUTH=specified` parameter. For
+Incremental backups will be appended to the full backup with `BACKUP ... INTO LATEST IN {collection location}`. Your storage location will contain the incremental in the default `/incrementals` directory at the collection location's root. For more detail on collections, see [Backup Collections](take-full-and-incremental-backups.html#backup-collections).
 
-In the following example `/2021/12/21-142943.73` contains the full backup. The incremental backups (`144748.08/` and `144639.97/`) are appended as subdirectories to the full backup:
+To output more detail about the backups contained within a directory, see [View a list of the full and incremental backups in a specific full backup subdirectory](show-backup.html#view-a-list-of-the-full-and-incremental-backups-in-a-specific-full-backup-subdirectory).
```
ditto: im not sure this sentence is necessary

kathancox replied:
Yes I agree, in general I think that section had got overworked over the release... will remove the last three sentences.

msbutler commented on v22.1/restore.md:
```diff
@@ -232,32 +243,41 @@ The examples in this section use the **default** `AUTH=specified` parameter. For
 ~~~
 
-When you restore a backup, add the backup's subdirectory path (e.g. `/2021/12/21-142943.73`) to the `RESTORE` statement.
+When you want to [restore a specific backup](#restore-a-specific-backup), add the backup's subdirectory path (e.g., `/2021/12/21-142943.73`) to the `RESTORE` statement. For details on viewing the most recent backup, see [`SHOW BACKUP LATEST`](show-backup.html#show-the-most-recent-backup).
```
nit: `SHOW BACKUP FROM <subdir> in <collectionURI>`

msbutler commented on v22.1/restore.md:
```diff
@@ -232,32 +243,41 @@ The examples in this section use the **default** `AUTH=specified` parameter. For
-       |—— 20211221/
-           |—— 144748.08/
-           |—— 144639.97/
+See [Incremental backups with explicitly specified destinations](take-full-and-incremental-backups.html#incremental-backups-with-explicitly-specified-destinations) to control where your backups go.
```
ditto: not sure a reference to incremental_from is needed here.

msbutler commented on v22.1/restore.md:
```diff
@@ -344,7 +373,7 @@ By default, tables and views are restored to the database they originally belong
 
 {% include_cached copy-clipboard.html %}
 ~~~ sql
-> RESTORE bank.customers FROM '2021/03/23-213101.37' IN 's3://{bucket_name}/{path/to/backup}?AWS_ACCESS_KEY_ID={key_id}&AWS_SECRET_ACCESS_KEY={access_key}'
```
tough that you have to copy and paste all this stuff for each external storage format! I wonder if it's even neccesary to have examples for each format? perhaps a seperate issue

kathancox replied:
Yes... too true. I have a docs issue to rework some markdown liquid template magic post release work. Apologies for the triple review.

msbutler commented on v22.1/restore.md:
```diff
@@ -283,38 +303,47 @@ To restore a single table:
  `RESTORE` will re-validate [indexes](indexes.html) when [incremental backups](take-full-and-incremental-backups.html) are created from an older version (v20.2.2 and earlier or v20.1.4 and earlier), but restored by a newer version (v21.1.0+). These earlier releases may have included incomplete data for indexes that were in the process of being created.
 {{site.data.alerts.end}}
 
+<span class="version-tag">New in v22.1:</span> To restore an incremental backup that was taken using the [`incremental_location` option](backup.html#incr-location), you must run the `RESTORE` statement with the full backup's location and the `incremental_location` option referencing the location passed in the original `BACKUP` statement:
```
as i said in other parts of this pr, i think an explanation of incremental_location, should be in the advanced use section

kathancox replied:
I am moving to include this in the advanced examples list on the backup page, and then including this on the `RESTORE` page but in the "Other restore usage" section.

msbutler commented on v21.2/take-full-and-incremental-backups.md:
```diff
@@ -40,36 +80,45 @@ To do a cluster backup, use the [`BACKUP`](backup.html) statement:
 {% include_cached copy-clipboard.html %}
 ~~~ sql
-> BACKUP INTO '{destination}';
+> BACKUP INTO '{collection location}';
```
heh, i think a discussion will clear up this syntax vagueness

msbutler commented on v22.1/take-and-restore-locality-aware-backups.md:
```diff
@@ -40,11 +40,13 @@ You can restore the backup by running:
 
 Note that the first URI in the list has to be the URI specified as the `default` URI when the backup was created. If you have moved your backups to a different location since the backup was originally taken, the first URI must be the new location of the files originally written to the `default` location.
 
+To restore from a specific backup, use [`RESTORE FROM {subdirectory} IN ...`](restore.html#restore-a-specific-backup)
```
i think this page could use some examples which use the incremental_location parameter, which also takes locality aware URIs. That change could also wait for another pre 22.1 pr, as I['m still figuring out](https://github.com/cockroachdb/cockroach/pull/79121) locality aware SHOW BACKUP.

kathancox replied:
I will cover this in a follow-up PR, with a separate docs issue that should hit my triage soon

msbutler reviewed (commented):
this is a monumental task! left a first round of comments. focused on backup.md, restore.md, and show-backup.md. I added two open questions to discussion tomorrow that I think will guide this pr to completion.

benbardin commented on _includes/cockroachcloud/restore-examples.md:
```diff
@@ -16,7 +16,7 @@
 ~~~
 
-When you restore a backup, add the backup's subdirectory path (e.g. `/2021/12/21-142943.73`) to the `RESTORE` statement.
+When you want to [restore a specific backup](../{{site.versions["stable"]}}/restore.html#restore-a-specific-backup), add the backup's subdirectory path (e.g., `/2021/12/21-142943.73`) to the `RESTORE` statement. For details on using `LATEST` with `SHOW BACKUP` to view the most recent backup, see [`SHOW BACKUP LATEST`](../{{site.versions["stable"]}}/show-backup.html#show-the-most-recent-backup).
```
That makes sense to me, we probably want to highlight LATEST as the primary restore mechanism.

benbardin commented on v22.1/backup.md:
```diff
@@ -58,7 +59,7 @@ To view the contents of an Enterprise backup created with the `BACKUP` statement
 -----------+-------------
 `targets` | Back up the listed [targets](#targets).
-`subdirectory` | The name of the specific subdirectory (e.g., `2021/03/23-213101.37`) where you want to add an [incremental backup](take-full-and-incremental-backups.html#incremental-backups). To view available subdirectories, use [`SHOW BACKUPS IN destination`](show-backup.html). If the `subdirectory` is not provided, a [full backup](take-full-and-incremental-backups.html#full-backups) will be c ...
+`subdirectory` | The name of the specific backup (e.g., `2021/03/23-213101.37`) in the collection to which you want to add an [incremental backup](take-full-and-incremental-backups.html#incremental-backups). To view available backup subdirectories, use [`SHOW BACKUPS IN destination`](show-backup.html). If the backup `subdirectory` is not provided, a [full backup](take-full-and-incremental-backups ...
```
I would love to see an overarching intro to backup/restore/show terminology, though it wouldn't necessarily fit in this specific file.

kathancox replied:
YES! I think there is room for a little more definition in this PR since I have introduced the Backup Collection section. But, I am in the process of writing a "Backup/Restore" overview page (post 22.1 GA release) — a glossary for terminology and listings of our different features (and in what products) is what I envision!

kathancox commented:
Updated per feedback. 

A couple of things to note: 

- backup/restore diagrams v21.2 & v21.1 need to be updated in a separate PR so the parameters have not been updated in those docs yet.
- The show backup v22.1 diagram update PR work is currently in progress, so have updated the 22.1 parameters as this will be up in the minor release. 
- For the sake of not broadening this PR any further, it is covering the issues/epics mentioned in the description and further 22.1 release issues will follow shortly based off this work, once merged.

msbutler commented on v22.1/backup.md:
```diff
@@ -56,13 +56,16 @@ To view the contents of an Enterprise backup created with the `BACKUP` statement
 -----------+-------------
 `targets` | Back up the listed [targets](#targets).
-`subdirectory` | The name of the specific backup (e.g., `2021/03/23-213101.37`) in the collection to which you want to add an [incremental backup](take-full-and-incremental-backups.html#incremental-backups). To view available backup subdirectories, use [`SHOW BACKUPS IN destination`](show-backup.html). If the backup `subdirectory` is not provided, a [full backup](take-full-and-incremental-backups ...
+`subdirectory` | The name of the specific backup (e.g., `2021/03/23-213101.37`) in the collection to which you want to add an [incremental backup](take-full-and-incremental-backups.html#incremental-backups). To view available backup subdirectories, use [`SHOW BACKUPS IN collectionURI`](show-backup.html). If the backup `subdirectory` is not provided, a [full backup](take-full-and-incremental-backu ...
```
I'm not sure what's standard for docs writing, but should we refrain from you/yours and stick to third person?

msbutler commented on v22.1/take-full-and-incremental-backups.md:
```diff
@@ -150,36 +154,36 @@ Periodically run the [`BACKUP`][backup] command to take a full backup of your cl
 
 {{site.data.alerts.callout_info}}
-If there are existing incremental backups in the storage location taken with v21.2 or earlier, new incremental backups will add to the existing incremental directory, (e.g., `{collection location}/{subdirectory}`). This follows how [incremental backups](take-full-and-incremental-backups.html#take-an-incremental-backup) were stored by default in v21.2 and earlier. To back up using the prior behavi ...
+If there are existing incremental backups in the storage location taken with v21.2 or earlier, new incremental backups will add to the existing incremental directory, (e.g., `{collectionURI}/{subdirectory}`). This follows how [incremental backups](take-full-and-incremental-backups.html#take-an-incremental-backup) were stored by default in v21.2 and earlier. To back up using the prior behavior, se ...
```
I think this could be a little clearer. Something along the line of: "In v21.2 and earlier, incremental backups were stored in the same directory as their full backup (i.e. collectionURI/subdirectory). If an incremental backup command points to a subdirectory with incremental backups created in v21.2 and earlier, v22.1 will write the incremental backup to the v21.2 default location).

msbutler commented on v22.1/take-full-and-incremental-backups.md:
```diff
@@ -150,36 +154,36 @@ Periodically run the [`BACKUP`][backup] command to take a full backup of your cl
 If it's ever necessary, you can then use the [`RESTORE`][restore] statement to restore your cluster, database(s), and/or table(s). Restoring from incremental backups requires previous full and incremental backups.
 
-To restore from an incremental backup, stored in the default `/incrementals` collection directory, run:
+To restore from an incremental backup, stored in the default `/incrementals` collection subdirectory, run:
```
I think the phrase "restore from an incremental backup" is slightly misleading, as it implies we only restore the incremental backup and not the whole chain. 

why  not say something like:  "restore from the latest backup in the collection." and the for the next example, just say "restore from a specific backup in the collection"

msbutler commented on v22.1/take-full-and-incremental-backups.md:
```diff
@@ -190,27 +194,27 @@ To define a specific subdirectory in the collection to restore from:
 <span class="version-tag">New in v22.1:</span> To explicitly control where your incremental backups go, use the [`incremental_location`](backup.html#options) option.
 
-In the following examples, the `{full_backup_location}` specifies the collection location containing the full backup. The `{incremental_backup_location}` is the alternative location that you can store an incremental backup:
+In the following examples, the `{collectionURI}` specifies the storage location containing the full backup. The `{explicit_incrementalsURI}` is the alternative location that you can store an incremental backup:
```
is it worth adding an intro sentence to this subsection?:

 "By default, incremental backups are stored in the /incrementals subdirectory at the root of the collection. Advance users may want to store incremental backups in a different location."

kathancox replied:
Although we have the advanced example section, I kind of shy away from defining users as "advanced".... What do you think about defining the use case instead? 

"By default, incremental backups are stored in the `/incrementals` subdirectory at the root of the collection. However, there are some advanced cases where you may want to store incremental backups in a different storage location."

I also have a potential follow-up project to build out a short section in the cloud storage docs re use cases.

msbutler replied:
this looks good!

msbutler reviewed (commented):
so much progress!!! I only reviewed v22.1 changes

kathancox commented on v22.1/backup.md:
```diff
@@ -137,7 +140,7 @@ Cancel the backup      | [`CANCEL JOB`](cancel-job.html)
 
 {{site.data.alerts.callout_info}}
-The presence of the `BACKUP MANIFEST` file in the backup destination is an indicator that the backup job completed successfully.
+The presence of the `BACKUP MANIFEST` file in the backup collection's storage location is an indicator that the backup job completed successfully.
```
Note to @kathancox: (to other version's)

kathancox commented on v22.1/backup.md:
```diff
@@ -244,7 +237,7 @@ See [Name Resolution](sql-name-resolution.html) for more details on how naming h
 ### Create incremental backups
 
-If you back up to a destination already containing a [full backup](take-full-and-incremental-backups.html#full-backups), an incremental backup will be added to the default `/incrementals` directory at the root of the [collection](take-full-and-incremental-backups.html#backup-collections) storage location:
+If you back up to a collection URI already containing a [full backup](take-full-and-incremental-backups.html#full-backups), an incremental backup will be added to the default `/incrementals` directory at the root of the [collection](take-full-and-incremental-backups.html#backup-collections) storage location:
```
Yes, sounds good! Both of those examples, are already included here!

kathancox commented on v22.1/backup.md:
```diff
@@ -253,17 +246,19 @@ If you back up to a destination already containing a [full backup](take-full-and
 ~~~
 
-To explicitly control where you store your incremental backups, use the [`incremental_location`](backup.html#options) option. In the following example, the cloud storage bucket specifies the collection location containing the full backup. The value passed to `incremental_location` is the alternative location that the incremental backup will be stored:
+To store the backup in a specific subdirectory in the storage location:
```
Hmm, I understand what you mean. But for technical documentation, where possible, I feel it's best to stacking code blocks without any kind of introduction. While the paragraph above fully explains what's happening, the two short sentence before each statement just helps guide the reader/user. If the user, doesn't need the intro, they can skip. Ofc, if there is something not technically accurate lmk!

kathancox commented on v22.1/backup.md:
```diff
@@ -56,13 +56,16 @@ To view the contents of an Enterprise backup created with the `BACKUP` statement
 -----------+-------------
 `targets` | Back up the listed [targets](#targets).
-`subdirectory` | The name of the specific backup (e.g., `2021/03/23-213101.37`) in the collection to which you want to add an [incremental backup](take-full-and-incremental-backups.html#incremental-backups). To view available backup subdirectories, use [`SHOW BACKUPS IN destination`](show-backup.html). If the backup `subdirectory` is not provided, a [full backup](take-full-and-incremental-backups ...
+`subdirectory` | The name of the specific backup (e.g., `2021/03/23-213101.37`) in the collection to which you want to add an [incremental backup](take-full-and-incremental-backups.html#incremental-backups). To view available backup subdirectories, use [`SHOW BACKUPS IN collectionURI`](show-backup.html). If the backup `subdirectory` is not provided, a [full backup](take-full-and-incremental-backu ...
```
Note sure of help and error text style guidelines in engineering... But, we use 2nd person pronouns in the docs per the style guide. It helps to avoid falling into passive voice too often, which can obscure meaning in some cases. Also, it helps promote a friendly tone in the docs.

kathancox commented on v22.1/show-backup.md:
```diff
@@ -7,27 +7,33 @@ docs_area: reference.sql
-`SHOW BACKUP FROM LATEST IN location` | Show the most recent backup added in the given [`location`](backup.html#backup-file-urls). [See the example](#show-the-most-recent-backup)
-`SHOW BACKUP FROM SCHEMAS location` | Show the schema details of the backup in the given [`location`](backup.html#backup-file-urls). [See the example](#show-a-backup-with-schemas).
-`SHOW BACKUP subdirectory IN location` |  List the full and incremental backups that are stored in the given full backup's `subdirectory` within a [`location`](backup.html#backup-file-urls). [See the example](#show-details-for-scheduled-backups).
+`SHOW BACKUPS IN location` | List the backup paths in the given [collection URI](backup.html#backup-file-urls). [See the example](#view-a-list-of-the-available-full-backup-subdirectories).
```
Yes, thank you!

kathancox commented on v22.1/show-backup.md:
```diff
@@ -275,26 +226,26 @@ You will receive an error if there is a collection of backups in the storage loc
 
 - `SHOW BACKUPS IN y` statement to [view a list of the available full backup subdirectories](#view-a-list-of-the-available-full-backup-subdirectories).
-- `SHOW BACKUP x IN y` statement to [view a list of the full and incremental backups that are stored in a specific full backup's subdirectory](#view-a-list-of-the-full-and-incremental-backups-in-a-specific-full-backup-subdirectory).
+- `SHOW BACKUP FROM x IN y` statement to [view a list of the full and incremental backups that are stored in a specific full backup's subdirectory](#view-a-list-of-the-full-and-incremental-backups-in-a-specific-full-backup-subdirectory).
```
`FROM` is in there! Thanks for the catch here.

kathancox commented on v22.1/take-full-and-incremental-backups.md:
```diff
@@ -38,9 +41,10 @@ Collection:
+[`SHOW BACKUPS IN {collectionURI}`](show-backup.html#view-a-list-of-the-available-full-backup-subdirectories) will display a list of the full backup subdirectories at the collection's URI.
 
-Alternately, the following directories also constitute a backup collection. There are multiple backups in two separate storage locations. Each individual backup is a full backup and its related incremental backup(s). Despite using the [`incremental_location`](#incremental-backups-with-explicitly-specified-destinations) option to store the incremental backup in an alternative location, that increm ...
+Alternately, the following directories also constitute a backup collection. There are multiple backups in two separate collection URIs. Each individual backup is a full backup and its related incremental backup(s). Despite using the [`incremental_location`](#incremental-backups-with-explicitly-specified-destinations) option to store the incremental backup in an alternative location, that incremen ...
```
Perhaps, I think the current location is a good central reference point for this description... so I've added an anchor link in from the advanced section, in case a reader wants more information on how this works.

kathancox commented:
Thanks @msbutler, I've incorporated the changes. Let me know if you have any other feedback/comments, otherwise this will move on to a docs writer for their review.

msbutler reviewed (approved):
LGTM! I didn't look as closely at the changes in non 22.1 pages, but in 22.1, I think everything is explained correctly!

kathancox commented:
> LGTM! I didn't look as closely at the changes in non 22.1 pages, but in 22.1, I think everything is explained correctly!

OK, thanks @msbutler. For 21.1 & 21.2, I only added the collections section per 22.1 minus the version-relevant features and updated the example syntax on `RESTORE` & `SHOW BACKUP` to include `LATEST`. There will be a docs review of this. And as soon as this is merged, I'll get the language in the diagrams and through the docs updated for 21.1 & 21.2. 

Thanks for reviewing this monster of a PR. It'll be great to have all of the terminology squared away with this PR and the next one!

andf-crl reviewed (commented):
Wow, amazing! Great work throughout, I have mostly just nits and language ideas. I may have suggested some changes to examples that you have already decided to hold off on per your message concerning updated SQL diagrams -- apologies if so -- please disregard! As always, all nits are optional! Great work!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/13448#-:-N028XbT-44OV_SfJwB2:b-qid4pv)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @kathancox, @livlobo, and @msbutler)

___
*[_includes/cockroachcloud/restore-examples.md, line 19 at r4](https://reviewable.io/reviews/cockroachdb/docs/13448#-N01QUvl21OuENp0cgj5:-N01QUvl21OuENp0cgj6:bhsn50w) ([raw file](https://github.com/cockroachdb/docs/blob/e6946e39d160489500a68d45d5b6b2623d19e7c7/_includes/cockroachcloud/restore-examples.md#L19)):*
> ```Markdown
> ~~~
> 
> To view the available subdirectories to restore a backup, use [`SHOW BACKUPS`](../{{site.versions["stable"]}}/restore.html#view-the-backup-subdirectories).
> ```

nit: to restore a backup from

or

from which to restore a backup

or, from `SHOW BACKUPS`:

To view the backup paths in a given destination

x3 on this page



_Code quote:_
```Markdown
to restore a backup
```
___
*[_includes/v22.1/backups/backup-to-deprec.md, line 2 at r4](https://reviewable.io/reviews/cockroachdb/docs/13448#-N01SRPkB6dtm5PzRIOO:-N01SRPkB6dtm5PzRIOP:b8msdrz) ([raw file](https://github.com/cockroachdb/docs/blob/e6946e39d160489500a68d45d5b6b2623d19e7c7/_includes/v22.1/backups/backup-to-deprec.md#L2)):*
> ```Markdown
> {{site.data.alerts.callout_danger}}
> The [`BACKUP ... TO`](../v20.2/backup.html) and [`RESTORE ... FROM`](../v20.2/restore.html) syntax is **deprecated** as of v22.1 and will be removed in a future release.
> ```

Q: Just wanted to clarify: Is `20.2` correct for both of these links? "Deprecated in 22.1" suggests that the last-supported version `21.2` might be appropriate here. However I see that `BACKUP ... TO` AND `RESTORE FROM` are each no longer mentioned starting in v21.1 and beyond (on `backup.html` and `restore.html` at least), so I think I understand the v20.2 links. Just commenting in case!
___
*[_includes/v22.1/backups/backup-to-deprec.md, line 4 at r4](https://reviewable.io/reviews/cockroachdb/docs/13448#-N01SxSfEV8kDMYHn1LY:-N01SxSfEV8kDMYHn1LZ:bbh90xs) ([raw file](https://github.com/cockroachdb/docs/blob/e6946e39d160489500a68d45d5b6b2623d19e7c7/_includes/v22.1/backups/backup-to-deprec.md#L4)):*
> ```Markdown
> The [`BACKUP ... TO`](../v20.2/backup.html) and [`RESTORE ... FROM`](../v20.2/restore.html) syntax is **deprecated** as of v22.1 and will be removed in a future release.
> 
> We recommend using the `BACKUP ... INTO {collection}` syntax, which creates or adds to a [backup collection](take-full-and-incremental-backups.html#backup-collections) in your storage location. For restoring backups, we recommend using `RESTORE FROM {backup} IN {collection}` with `{backup}` being `LATEST` or a specific subdirectory.
> ```

nit: What do you think of linking the two salient options for users here:

LATEST: [`LATEST`](../{{site.versions["stable"]}}/restore.html#restore-the-most-recent-backup).
subdirectory: (create an anchor link for the `subdirectory` parameters reference definition here: v22.1/backup.html#parameters, then link to it.

There are ~12 mentions following the above pattern in this PR: no need to link them all in this fashion, but the first instance on-page for each page would be ideal if possible!
___
*[v21.1/backup.md, line 69 at r4](https://reviewable.io/reviews/cockroachdb/docs/13448#-N027dAuACjv5e9T3ess:-N027dAuACjv5e9T3est:bkdikjh) ([raw file](https://github.com/cockroachdb/docs/blob/e6946e39d160489500a68d45d5b6b2623d19e7c7/v21.1/backup.md#L69)):*
> ```Markdown
> `targets` | Back up the listed [targets](#targets).
> `subdirectory` | The name of the specific backup (e.g., `2021/03/23-213101.37`) in the collection to which you want to add an [incremental backup](take-full-and-incremental-backups.html#incremental-backups). To view available backup subdirectories, use [`SHOW BACKUPS IN destination`](show-backup.html). If the backup `subdirectory` is not provided, a [full backup](take-full-and-incremental-backups.html#full-backups) will be created in the collection using a date-based naming scheme (i.e., `<year>/<month>/<day>-<timestamp>`).<br><br>**Warning:** If you use an arbitrary `STRING` as the subdirectory, a new full backup will be created, but it will never be shown in `SHOW BACKUPS IN`. We do not recommend using arbitrary strings as subdirectory names.
> `LATEST` | Append an incremental backup to the latest completed full backup's subdirectory.
> ```

I think it would be valuable to add here that the default location for these incrementals is `/incrementals`, since this is a behavior change from 21.2. Perhaps also useful to briefly mention `incremental_location`? Applies to each version of BACKUP (21.1, 21.2, 22.1)
___
*[v21.1/restore.md, line 55 at r4](https://reviewable.io/reviews/cockroachdb/docs/13448#-N01a8wk-sNSRVq2CoTX:-N01a8wk-sNSRVq2CoTY:b99bj7h) ([raw file](https://github.com/cockroachdb/docs/blob/e6946e39d160489500a68d45d5b6b2623d19e7c7/v21.1/restore.md#L55)):*
> ```Markdown
>  `database_name` | The name of the database you want to restore (i.e., restore all tables and views in the database). You can restore an entire database only if you had backed up the entire database.
>  `destination` | The URL where the [full backup](take-full-and-incremental-backups.html#full-backups) (and appended [incremental backups](take-full-and-incremental-backups.html#incremental-backups), if applicable) is stored. <br/><br/>For information about this URL structure, see [Backup File URLs](#backup-file-urls).
>   `LATEST` | Restore the most recent backup in the given location. See the [Restore from the most recent backup](#restore-the-most-recent-backup) example.
> ```

Q: Should there also be a `subdirectory` entry here in parameters, similar to that in backup.html? I'm not certain, but it seems that there should be given its presence in the SQL diagram.
___
*[v21.1/restore.md, line 250 at r4](https://reviewable.io/reviews/cockroachdb/docs/13448#-N01Z88l-r5osVUjdC4X:-N01Z88l-r5osVUjdC4Y:b-v1jqvo) ([raw file](https://github.com/cockroachdb/docs/blob/e6946e39d160489500a68d45d5b6b2623d19e7c7/v21.1/restore.md#L250)):*
> ```Markdown
> ### Restore the most recent backup
> 
> <span class="version-tag">New in v21.1:</span> To restore from the most recent backup in the collection's location, use the `LATEST syntax`:
> ```

Q: Technically, if I'm running v21.1.13, this feature is not a new feature for me, it is in fact a nonexistant feature for me: new in v21.1.14, not new in v21.1.0. I think, however, that this is site-standard here, so no action needed -- just wanted to raise for awareness. This is always a complication with upstream backports: how to effectively communicate the change to previous versions.
___
*[v21.1/restore.md, line 259 at r4](https://reviewable.io/reviews/cockroachdb/docs/13448#-N01_oupDfOg96Yc27lQ:-N01_oupDfOg96Yc27lR:bjpt4ch) ([raw file](https://github.com/cockroachdb/docs/blob/e6946e39d160489500a68d45d5b6b2623d19e7c7/v21.1/restore.md#L259)):*
> ```Markdown
> ### Restore a specific backup
> 
> To restore a specific backup, use the backup's subdirectory in the collection's location:
> ```

nit: subdirectory could also be linked to a new anchor tag for this.
___
*[v21.1/restore.md, line 274 at r4](https://reviewable.io/reviews/cockroachdb/docs/13448#-N01l_eL4DUAq0vLyGz0:-N01l_eL4DUAq0vLyGz1:b-v9u5pn) ([raw file](https://github.com/cockroachdb/docs/blob/e6946e39d160489500a68d45d5b6b2623d19e7c7/v21.1/restore.md#L274)):*
> ```Markdown
> {% include_cached copy-clipboard.html %}
> ~~~ sql
> > RESTORE FROM LATEST IN 's3://{bucket_name}/{path/to/backup}?AWS_ACCESS_KEY_ID={key_id}&AWS_SECRET_ACCESS_KEY={access_key}';
> ```

Q: did you want to remove `/{path/to/backup}` from code examples on this page also? 25 instances on this page.
___
*[v21.1/take-and-restore-encrypted-backups.md, line 102 at r4](https://reviewable.io/reviews/cockroachdb/docs/13448#-N01sMoWCrU4wEa7iJHb:-N01sMoWCrU4wEa7iJHc:b-bqpjgo) ([raw file](https://github.com/cockroachdb/docs/blob/e6946e39d160489500a68d45d5b6b2623d19e7c7/v21.1/take-and-restore-encrypted-backups.md#L102)):*
> ```Markdown
> {% include_cached copy-clipboard.html %}
> ~~~ sql
> > RESTORE FROM LATEST IN 's3://{BUCKET NAME}/{PATH}?AWS_ACCESS_KEY_ID={KEY ID}&AWS_SECRET_ACCESS_KEY={SECRET ACCESS KEY}'
> ```

Q: Did you want to remove `/{PATH}` on this page also? 4 instances on-page.
___
*[v21.1/take-full-and-incremental-backups.md, line 72 at r4](https://reviewable.io/reviews/cockroachdb/docs/13448#-N01gIBz8Jw39Y2JZenI:-N01gIBz8Jw39Y2JZenJ:b-9bxwto) ([raw file](https://github.com/cockroachdb/docs/blob/e6946e39d160489500a68d45d5b6b2623d19e7c7/v21.1/take-full-and-incremental-backups.md#L72)):*
> ```Markdown
> ### Take a full backup
> 
> To do a cluster backup, use the [`BACKUP`](backup.html) statement:
> ```

inherited nit: "To perform a full cluster backup, use the ..."



_Code quote:_
```Markdown
To do a cluster backup
```
___
*[v21.1/take-full-and-incremental-backups.md, line 79 at r4](https://reviewable.io/reviews/cockroachdb/docs/13448#-N01gP0D8TvMzHuKLMUO:-N01gP0D8TvMzHuKLMUP:b-lc6nnl) ([raw file](https://github.com/cockroachdb/docs/blob/e6946e39d160489500a68d45d5b6b2623d19e7c7/v21.1/take-full-and-incremental-backups.md#L79)):*
> ```Markdown
> ~~~
> 
> If it's ever necessary, you can use the [`RESTORE`][restore] statement with `LATEST` to restore the most recent backup added to the [collection]:
> ```

inherited nit: "To restore a backup, use the ..." Here, also, we can be general, since the guidance below is broken out by specific purpose. So perhaps something like:

"""
To restore a backup, use the [`RESTORE`][restore] statement, specifying what you want to restore as well as  the `[collection](take-full-and-incremental-backups.html#backup-collections)`:

- To restore the latest backup of a table:
- To restore the latest backup of a database:
- To restore the latest backup of your full cluster:
- To restore a backup from a specific subdirectory:

"""

What do you think of converting the "to restore a X" lines into unordered list elements in this fashion? This also breaks LATEST out of the intro sentence, for which item #4: specific subdir doesn't apply.

Guess: `[collection]` should probably be`[collection](take-full-and-incremental-backups.html#backup-collections)`



_Code quote:_
```Markdown
If it's ever necessary
```
___
*[v21.2/restore.md, line 410 at r4](https://reviewable.io/reviews/cockroachdb/docs/13448#-N01l3n6AR6vNjCQ72aN:-N01l3n710oCvzHVtzSs:b-vah2qy) ([raw file](https://github.com/cockroachdb/docs/blob/e6946e39d160489500a68d45d5b6b2623d19e7c7/v21.2/restore.md#L410)):*
> ```Markdown
> {% include_cached copy-clipboard.html %}
> ~~~ sql
> > RESTORE system.users  FROM LATEST IN 's3://{bucket_name}?AWS_ACCESS_KEY_ID={key_id}&AWS_SECRET_ACCESS_KEY={access_key}'
> ```

supernit: extra space here: `system.users  FROM`
___
*[v21.2/take-and-restore-encrypted-backups.md, line 103 at r4](https://reviewable.io/reviews/cockroachdb/docs/13448#-N01sWeW5foYBpAaEN9P:-N01sWeW5foYBpAaEN9Q:bii3pw1) ([raw file](https://github.com/cockroachdb/docs/blob/e6946e39d160489500a68d45d5b6b2623d19e7c7/v21.2/take-and-restore-encrypted-backups.md#L103)):*
> ```Markdown
> {% include_cached copy-clipboard.html %}
> ~~~ sql
> > RESTORE FROM LATEST IN 's3://{BUCKET NAME}/{PATH}?AWS_ACCESS_KEY_ID={KEY ID}&AWS_SECRET_ACCESS_KEY={SECRET ACCESS KEY}'
> ```

Same here: did you want to remove `/{PATH}` on this page also? 4 instances on-page.
___
*[v21.2/take-full-and-incremental-backups.md, line 90 at r4](https://reviewable.io/reviews/cockroachdb/docs/13448#-N01o6xcBnwsEB3QyGKn:-N01o6xcBnwsEB3QyGKo:b-yyy3z2) ([raw file](https://github.com/cockroachdb/docs/blob/e6946e39d160489500a68d45d5b6b2623d19e7c7/v21.2/take-full-and-incremental-backups.md#L90)):*
> ```Markdown
> ~~~
> 
> If it's ever necessary, you can use the [`RESTORE`][restore] statement with `LATEST` to restore the most recent backup added to the [collection]:
> ```

nit: same ordering recommendation as with the v21.1 version of this page: consider UL elements for specific restore targets.
___
*[v21.2/take-full-and-incremental-backups.md, line 162 at r4](https://reviewable.io/reviews/cockroachdb/docs/13448#-N01oJ1sENg_APCuBazA:-N01oJ1t9IQXZBh3GYwN:b-8kf8n7) ([raw file](https://github.com/cockroachdb/docs/blob/e6946e39d160489500a68d45d5b6b2623d19e7c7/v21.2/take-full-and-incremental-backups.md#L162)):*
> ```Markdown
> If it's ever necessary, you can then use the [`RESTORE`][restore] command to restore your cluster, database(s), and/or table(s). Restoring from incremental backups requires previous full and incremental backups.
> 
> To restore from the most recent incremental backup, run the following:
> ```

Q: wouldn't this restore the most recent backup of any kind (incremental|full) ? Does it make sense then to be explicit here, and later with `To restore from a specific incremental backup ...`

If not, I am confused because "Restore a Database" and "Restore from incremental backups" use identical code examples over here:

https://deploy-preview-13448--cockroachdb-docs.netlify.app/docs/dev/restore.html#restore-a-database

https://deploy-preview-13448--cockroachdb-docs.netlify.app/docs/dev/restore.html#restore-from-incremental-backups
___
*[v22.1/backup.md, line 68 at r4](https://reviewable.io/reviews/cockroachdb/docs/13448#-N01qW6t3xiTRS8_MQJB:-N01qW6t3xiTRS8_MQJC:b-qvraaq) ([raw file](https://github.com/cockroachdb/docs/blob/e6946e39d160489500a68d45d5b6b2623d19e7c7/v22.1/backup.md#L68)):*
> ```Markdown
> `collectionURI` | The URI where you want to store the backup. (Or, the default locality for a locality-aware backup.)<br/><br/>For information about this URL structure, see [Backup File URLs](#backup-file-urls).
> `localityURI`   | The URI containing the `COCKROACH_LOCALITY` parameter for a non-default locality that is part of a single locality-aware backup.
> `timestamp` | Back-up data as it existed as of [`timestamp`](as-of-system-time.html). The `timestamp` must be more recent than your cluster's last garbage collection (which defaults to occur every 25 hours, but is [configurable per table](configure-replication-zones.html#replication-zone-variables)).
> ```

nit: "back up" appears correct



_Code quote:_
```Markdown
Back-up
```
___
*[v22.1/backup.md, line 255 at r4](https://reviewable.io/reviews/cockroachdb/docs/13448#-N01sgD_-WxdSSOgDbI8:-N01sgD_-WxdSSOgDbI9:b-oh5oda) ([raw file](https://github.com/cockroachdb/docs/blob/e6946e39d160489500a68d45d5b6b2623d19e7c7/v22.1/backup.md#L255)):*
> ```Markdown
> {% include_cached copy-clipboard.html %}
> ~~~ sql
> BACKUP DATABASE bank INTO {'subdirectory'} IN 's3://{BUCKET NAME}/{PATH}?AWS_ACCESS_KEY_ID={KEY ID}&AWS_SECRET_ACCESS_KEY={SECRET ACCESS KEY}' \
> ```

Q: Same here: did you want to remove `/{PATH}` on this page also? 6 instances on-page.
___
*[v22.1/backup.md, line 263 at r4](https://reviewable.io/reviews/cockroachdb/docs/13448#-N01sxaXBh32fMQ00INM:-N01sxaXBh32fMQ00INN:b-13h6zp) ([raw file](https://github.com/cockroachdb/docs/blob/e6946e39d160489500a68d45d5b6b2623d19e7c7/v22.1/backup.md#L263)):*
> ```Markdown
> {{site.data.alerts.end}}
> 
> To explicitly control where you store your incremental backups, use the [`incremental_location`](backup.html#options) option. For more detail, see [this example](take-full-and-incremental-backups.html#incremental-backups-with-explicitly-specified-destinations) on using the `incremental_location` option.
> ```

supernit: `on using` --> `demonstrating`
___
*[v22.1/backup.md, line 393 at r4](https://reviewable.io/reviews/cockroachdb/docs/13448#-N01tRCr7vDSeQXNk2pw:-N01tRCr7vDSeQXNk2px:bvla704) ([raw file](https://github.com/cockroachdb/docs/blob/e6946e39d160489500a68d45d5b6b2623d19e7c7/v22.1/backup.md#L393)):*
> ```Markdown
> ~~~ sql
> BACKUP DATABASE bank INTO 'subdirectory' IN 'azure://{CONTAINER NAME}/{PATH}?AZURE_ACCOUNT_NAME={ACCOUNT NAME}&AZURE_ACCOUNT_KEY={URL-ENCODED KEY}' \
> AS OF SYSTEM TIME '-10s';
> ```

supernit: for code examples that showcase small differences between usage (i.e `LATEST` vs `'subdirectory'`), it's ideal to keep everything else about the example the same to better focus on the difference. For example, these two examples showcase a cluster backup vs a specific database backup. If `LATEST` and `'subdirectory'` otherwise function identically between the two, let's standardize the two examples to use the same backup target:

- `BACKUP INTO LATEST IN ...`
- `BACKUP INTO 'subdirectory' IN ...`
___
*[v22.1/backup.md, line 539 at r4](https://reviewable.io/reviews/cockroachdb/docs/13448#-N01vY9T74eC-_5Dsfwm:-N01vY9YA9biip-QPv2K:b-cwuvc2) ([raw file](https://github.com/cockroachdb/docs/blob/e6946e39d160489500a68d45d5b6b2623d19e7c7/v22.1/backup.md#L539)):*
> ```Markdown
> {{site.data.alerts.end}}
> 
> To explicitly control where you store your incremental backups, use the [`incremental_location`](backup.html#options) option. For more detail, see [this example](take-full-and-incremental-backups.html#incremental-backups-with-explicitly-specified-destinations) on using the `incremental_location` option including how to restore incremental backups taken with it.
> ```

nit: `on using` --> `demonstrating` or `showcasing`

nit: `option including how` --> `option and how`

or similar
___
*[v22.1/create-schedule-for-backup.md, line 225 at r4](https://reviewable.io/reviews/cockroachdb/docs/13448#-N01wNfJEmqRO7d7PQJ9:-N01wNfJEmqRO7d7PQJA:btd6g2u) ([raw file](https://github.com/cockroachdb/docs/blob/e6946e39d160489500a68d45d5b6b2623d19e7c7/v22.1/create-schedule-for-backup.md#L225)):*
> ```Markdown
> 
> - `SHOW BACKUPS IN y` statement to [view a list of the full backup's subdirectories](show-backup.html#view-a-list-of-the-available-full-backup-subdirectories).
> - `SHOW BACKUP FROM x IN y` statement to [view a list of the full and incremental backups that are stored in a specific full backup's subdirectory](show-backup.html#view-a-list-of-the-full-and-incremental-backups-in-a-specific-full-backup-subdirectory).
> ```

nit: should these be updated to match elsewhere:

- `SHOW BACKUPS IN collectionURI` statement ...
- `SHOW BACKUP FROM subdirectory IN collectionURI` statement ...
___
*[v22.1/manage-a-backup-schedule.md, line 68 at r4](https://reviewable.io/reviews/cockroachdb/docs/13448#-N01wWzDDxfg9QahTAh4:-N01wWzDDxfg9QahTAh5:bsyah4w) ([raw file](https://github.com/cockroachdb/docs/blob/e6946e39d160489500a68d45d5b6b2623d19e7c7/v22.1/manage-a-backup-schedule.md#L68)):*
> ```Markdown
> 
> - `SHOW BACKUPS IN y` statement to [view a list of the full backup's subdirectories](show-backup.html#view-a-list-of-the-available-full-backup-subdirectories).
> - `SHOW BACKUP FROM x IN y` statement to [view a list of the full and incremental backups that are stored in a specific full backup's subdirectory](show-backup.html#view-a-list-of-the-full-and-incremental-backups-in-a-specific-full-backup-subdirectory).
> ```

nit: same, should these be updated to match elsewhere:

- `SHOW BACKUPS IN collectionURI` statement ...
- `SHOW BACKUP FROM subdirectory IN collectionURI` statement ...
___
*[v22.1/restore.md, line 225 at r4](https://reviewable.io/reviews/cockroachdb/docs/13448#-N01xFaE6H8OPnGVRYpm:-N01xFaE6H8OPnGVRYpn:b-bl40ns) ([raw file](https://github.com/cockroachdb/docs/blob/e6946e39d160489500a68d45d5b6b2623d19e7c7/v22.1/restore.md#L225)):*
> ```Markdown
> - [Restoring from a specific backup](#restore-from-a-specific-backup)
> 
> The examples in this section use the `LATEST` syntax; however, you can also [restore a specific directory](#restore-a-specific-backup).
> ```

nit: UL element 2 ('specific')is already exactly this 'however' clause. Do we need them both?

I agree that it is helpful to advise the user that we'll be using LATEST throughout our examples, but it would be more helpful if we connected "most recent backup" from UL 1 to "LATEST" in follow-up sentence. Perhaps:

"""
The examples in this section demonstrate restoring from the most recent backup using the `LATEST` syntax.
"""
___
*[v22.1/restore.md, line 448 at r4](https://reviewable.io/reviews/cockroachdb/docs/13448#-N01y6FC1Her48KI540l:-N01y6FC1Her48KI540m:b-f7bdn2) ([raw file](https://github.com/cockroachdb/docs/blob/e6946e39d160489500a68d45d5b6b2623d19e7c7/v22.1/restore.md#L448)):*
> ```Markdown
> ### Restore from the most recent backup
> 
> <span class="version-tag">New in v22.1:</span> To restore from the most recent backup in the collection's location, use the `LATEST syntax`:
> ```

nit: `LATEST syntax` --> `LATEST` syntax
___
*[v22.1/restore.md, line 602 at r4](https://reviewable.io/reviews/cockroachdb/docs/13448#-N01yfUqCULQopomxv7B:-N01yfUqCULQopomxv7C:b-h0xa62) ([raw file](https://github.com/cockroachdb/docs/blob/e6946e39d160489500a68d45d5b6b2623d19e7c7/v22.1/restore.md#L602)):*
> ```Markdown
> #### Restore from incremental backups in a different location
> 
> <span class="version-tag">New in v22.1:</span> To restore an incremental backup that was taken using the [`incremental_location` option](backup.html#incr-location), you must run the `RESTORE` statement with the full backup's collection location URI and the `incremental_location` option referencing the collection URI passed in the original `BACKUP` statement:
> ```

nit: This sentence is a little difficult to parse. What do you think about breaking this out to UL elements:

"""
To restore an incremental backup that was taken using the [`incremental_location` option](backup.html#incr-location), you must run the `RESTORE` statement with both:

- the collection URI of the full backup
- the `incremental_location` option referencing the incremental backup's collection URI, as passed in the original `BACKUP` statement
"""
___
*[v22.1/restore.md, line 621 at r4](https://reviewable.io/reviews/cockroachdb/docs/13448#-N01zhCy-jKHiVAUIyJc:-N01zhD14V-552c8Jcv6:b-fwkg6x) ([raw file](https://github.com/cockroachdb/docs/blob/e6946e39d160489500a68d45d5b6b2623d19e7c7/v22.1/restore.md#L621)):*
> ```Markdown
> ### View the backup subdirectories
> 
> `BACKUP ... INTO` adds a backup to a [backup collection] location. To view the backup paths in a given collection location, use [`SHOW BACKUPS`](show-backup.html):
> ```

This just needs a link like:

[backup collection](take-full-and-incremental-backups.html#backup-collections)
___
*[v22.1/show-backup.md, line 138 at r4](https://reviewable.io/reviews/cockroachdb/docs/13448#-N020SHNBIcGCFAQCEYb:-N020SHNBIcGCFAQCEYc:bqofvle) ([raw file](https://github.com/cockroachdb/docs/blob/e6946e39d160489500a68d45d5b6b2623d19e7c7/v22.1/show-backup.md#L138)):*
> ```Markdown
> system        | public             | role_members               | table       | incremental | 2022-04-08 14:23:55.33557 | 2022-04-08 14:26:01.699694 |         94 |     1 |      true
> . . .
> ~~~
> ```

nit: it's tough to see the difference between:
- Show the most recent backup
- View a list of the full and incremental backups in a specific full backup subdirectory

Results output appears very similar, and command differs only by LATEST vs supplied date. Both show full and incremental backups, the latter just has an additional backup listed. Not sure if easily addressed, and merely a nit!! -- but would be helpful if the results output showcased how the two uses of SHOW BACKUP differ.
___
*[v22.1/take-and-restore-encrypted-backups.md, line 164 at r1](https://reviewable.io/reviews/cockroachdb/docs/13448#-MzX16NG_8nhWKjQnpGa-r1-164:-N021yBzD3EeATCqtaB7:bon11jl) ([raw file](https://github.com/cockroachdb/docs/blob/fe9076b84bc483d60e568516b1d5a00da756fcad/v22.1/take-and-restore-encrypted-backups.md#L164)):*
<details><summary><i>Previously, msbutler (Michael Butler) wrote…</i></summary><blockquote>

nit: remove "specific"? a user may get the wrong idea that they can restore a single incremental backup or something.
</blockquote></details>

nit: if possible, standardize this phrasing throughout this PR, whichever is the preferred approach:

- 3 instances: To restore from a backup, use [`RESTORE FROM {subdirectory} IN ...`]
- 17 instances: To restore from a specific backup, use [`RESTORE FROM {subdirectory} IN ...`]

For my part, I think "specific" belongs.
___
*[v22.1/take-and-restore-encrypted-backups.md, line 201 at r4](https://reviewable.io/reviews/cockroachdb/docs/13448#-N01Oz4H6YrYaBSzOt7D:-N01Oz4H6YrYaBSzOt7E:bw3lp5h) ([raw file](https://github.com/cockroachdb/docs/blob/e6946e39d160489500a68d45d5b6b2623d19e7c7/v22.1/take-and-restore-encrypted-backups.md#L201)):*
> ```Markdown
> 
> {% include_cached copy-clipboard.html %}
> ~~~
> ```

nit: should this retain `sql` code highlighting to match others?
___
*[v22.1/take-backups-with-revision-history-and-restore-from-a-point-in-time.md, line 12 at r4](https://reviewable.io/reviews/cockroachdb/docs/13448#-N022ziG4GwOLRXq_hm7:-N022ziG4GwOLRXq_hm8:b-l2j3cx) ([raw file](https://github.com/cockroachdb/docs/blob/e6946e39d160489500a68d45d5b6b2623d19e7c7/v22.1/take-backups-with-revision-history-and-restore-from-a-point-in-time.md#L12)):*
> ```Markdown
> {{site.data.alerts.end}}
> 
> This doc provides information about how to take backups with revision history and restore from a point-in-time.
> ```

inherited nit: `doc` --> `page`
___
*[v22.1/take-backups-with-revision-history-and-restore-from-a-point-in-time.md, line 25 at r4](https://reviewable.io/reviews/cockroachdb/docs/13448#-N0232nf7LQv1WCkT72j:-N0232nf7LQv1WCkT72k:b8c910r) ([raw file](https://github.com/cockroachdb/docs/blob/e6946e39d160489500a68d45d5b6b2623d19e7c7/v22.1/take-backups-with-revision-history-and-restore-from-a-point-in-time.md#L25)):*
> ```Markdown
> {% include_cached copy-clipboard.html %}
> ~~~ sql
> > BACKUP INTO '{collection location}' AS OF SYSTEM TIME '-10s' WITH revision_history;
> ```

Q: should this be `'{collectionURI}'` to match elsewhere? Also present in next example.
___
*[v22.1/take-backups-with-revision-history-and-restore-from-a-point-in-time.md, line 40 at r4](https://reviewable.io/reviews/cockroachdb/docs/13448#-N023DmMBEt49F7JAo43:-N023DmMBEt49F7JAo44:b-j3cyxw) ([raw file](https://github.com/cockroachdb/docs/blob/e6946e39d160489500a68d45d5b6b2623d19e7c7/v22.1/take-backups-with-revision-history-and-restore-from-a-point-in-time.md#L40)):*
> ```Markdown
> {% include_cached copy-clipboard.html %}
> ~~~ sql
> > RESTORE FROM '/2021/12/13-211056.62' IN '{collection location}' AS OF SYSTEM TIME '2021-12-13 10:00:00';
> ```

Q (very unsure!): should this also be `LATEST` instead of the specific date to update our recommended approach where applicable?
___
*[v22.1/take-full-and-incremental-backups.md, line 170 at r2](https://reviewable.io/reviews/cockroachdb/docs/13448#-MzX16NHP4mK8pX1MxIr-r2-170:-N025zjj7oLFx-4Bx_-Q:be0mlax) ([raw file](https://github.com/cockroachdb/docs/blob/34637cee1ecb6f3bc3099131e40ae8fb61a78ca9/v22.1/take-full-and-incremental-backups.md#L170)):*
<details><summary><i>Previously, msbutler (Michael Butler) wrote…</i></summary><blockquote>

I think this could be a little clearer. Something along the line of: "In v21.2 and earlier, incremental backups were stored in the same directory as their full backup (i.e. collectionURI/subdirectory). If an incremental backup command points to a subdirectory with incremental backups created in v21.2 and earlier, v22.1 will write the incremental backup to the v21.2 default location). 
</blockquote></details>

nit: Recommend:

"""
To backup using the prior behavior, use the `incremental_location` option, as shown in [this example](#backup-earlier-behavior).
"
___
*[v22.1/take-full-and-incremental-backups.md, line 25 at r4](https://reviewable.io/reviews/cockroachdb/docs/13448#-N023qOyCG8xsMOCBA1E:-N023qOyCG8xsMOCBA1F:b-umkrfd) ([raw file](https://github.com/cockroachdb/docs/blob/e6946e39d160489500a68d45d5b6b2623d19e7c7/v22.1/take-full-and-incremental-backups.md#L25)):*
> ```Markdown
> ## Backup collections
> 
>  A _backup collection_ defines a set of backups and their metadata. The collection can contain multiple full backups and their subsequent [incremental backups](#incremental-backups). The path to a backup is created using a date-based naming scheme and stored at the URI passed with the `BACKUP` statement.
> ```

nit: should we link URI ("collection URI "per michael) to BACKUP's params entry for `collectionURI`, since this is its first mention? Might need an anchor link?
___
*[v22.1/take-full-and-incremental-backups.md, line 32 at r4](https://reviewable.io/reviews/cockroachdb/docs/13448#-N024PcT8wF1rdUK04YY:-N024PcT8wF1rdUK04YZ:b1zm3uo) ([raw file](https://github.com/cockroachdb/docs/blob/e6946e39d160489500a68d45d5b6b2623d19e7c7/v22.1/take-full-and-incremental-backups.md#L32)):*
> ```Markdown
> - As of v22.1, it is possible to store incremental backups at a [different URI](#incremental-backups-with-explicitly-specified-destinations) to the related full backup. This means that one or multiple storage locations can hold one backup collection.
> 
> In the following example, one storage bucket is holding the backup collection. The full backups are stored at the root of the collection's URI in a date-based path. The `/incrementals` directory holds all incremental backups by default.
> ```

nit: Let's lead with "this is default":

"""
By default, full backups are stored at the root of the collection's URI in a date-based path, and incremental backups are stored in the `/incrementals` directory. The following example shows a backup collection created using these default values, where all backups reside on one storage bucket:
"""
___
*[v22.1/take-full-and-incremental-backups.md, line 207 at r4](https://reviewable.io/reviews/cockroachdb/docs/13448#-N02-4XF-drWQ8FT4OgM:-N02-4XF-drWQ8FT4OgN:b-m6fj4y) ([raw file](https://github.com/cockroachdb/docs/blob/e6946e39d160489500a68d45d5b6b2623d19e7c7/v22.1/take-full-and-incremental-backups.md#L207)):*
> ```Markdown
> A full backup must be present in the `{collectionURI}` in order to take an incremental backup to the alternative `{explicit_incrementalsURI}`. If there isn't a full backup present in `{collectionURI}` when taking an incremental backup with `incremental_location`, the error `path does not contain a completed latest backup` will be returned.
> 
> For details on the backup directory structure when taking incremental backups with `incremental_location`, see [Backup collections](#incremental-location-structure)
> ```

nit: if `[backup collections]` is a different link target than the one two paragraphs above, it should have unique link text. Otherwise no one will click it, since they just read the one just above! I think the link target is correct here, just needs updated link text.


<!-- Sent from Reviewable.io -->

kathancox commented:
Hey @andf-crl Thanks so much for a great review. I think I pretty much incorporated your feedback. There were a couple questions you had:

- Yes, the 20.2 links in the deprecation seem weird. The reason we've directed there is because backup collections aren't a thing in 20.2. Although you can still use the old syntax (`BACKUP TO`) in 21.1+, we've updated the later versions to the new syntax to try and get folks switched over to our collection usage.
- One of your comments referred to mentioning the `/incrementals` default and incremental_location option in the parameters table. I have done this for 22.1 (the default dir), but this isn't a thing in 21.1 or 21.2. The option I haven't mentioned in the parameters table, since it is a pretty specific use case.
- The version tags! Yes, you're right. We used to only do that major version (or afaik), the guidance was updated recently: https://github.com/cockroachdb/docs/blob/master/StyleGuide.md#version-tags, so I've added the minors in for 21.1/2.
- The subdirectory/latest on the point in time restore... I haven't changed. I had a convo about this with an engineer and have opened another issue to clarify some explanation around there.

msbutler commented on v22.1/take-full-and-incremental-backups.md:
```diff
@@ -92,45 +153,72 @@ Periodically run the [`BACKUP`][backup] command to take a full backup of your cl
+<a name="backup-earlier-behavior"></a> To take incremental backups to a subdirectory in the `collectionURI` that you define (rather than the default `/incrementals` directory), use the `incremental_location` option:
+
+~~~ sql
+BACKUP INTO LATEST IN '{collectionURI}' AS OF SYSTEM TIME '-10s' WITH incremental_location = '{collectionURI/subdirectory}';
```
i wonder if this example should get moved to the "callout_info" blurb for pre 22.1 behavior? If you don't think the example should moved, perhaps mention that this example is for folks that want to mimic pre 22.1 behavior.

also the the incremental_location should be '{collectionURI}', instead of '{collectionURI/subdirectory}'. The LATEST keyword will handle the subdirectory resolution.

kathancox replied:
So if you want to mimic old behavior: `BACKUP INTO LATEST IN {collectionURI}` or `BACKUP INTO {subdirectory} in {collectionURI}`... isn't `BACKUP INTO LATEST IN '{collectionURI}' AS OF SYSTEM TIME '-10s' WITH incremental_location = '{collectionURI}'` .... if you don't specify a subdir on the second `collectionURI` (which is the same as the first because we're trying to mimic the old behavior), will `incremental_location` create a subdir that isn't `/incrementals` in v22.1?

kathancox replied:
OK so I tested again, and if you just specify the same `collectionURI` with no `/subdir` the incremental will just go into a date-based path under the full backup.... it won't go into the `/incrementals`. 

If you then run `BACKUP DATABASE movr INTO LATEST IN {collectionURI}` it will continue with the old behavior, it will won't then create a default `/incrementals` directory... I assume to maintain the chain in some way. Expected behavior? If so, I will add a sentence to this effect now.

msbutler replied:
yeah this is expected and real subtle:

If the backup already contains incrementals using the pre 21.2 style (i.e. inceremtals in the full backup dir), in 22.1 `BACKUP INTO LATEST IN '{collectionURI}' ` will store the incrementals by default in the  full dir. this is to maintain backward compatibility for users trying to maintain a backup chain across versions.

Now, if the user in 22.1 runs `BACKUP INTO LATEST IN '{collectionURI}' ` and the full backup does _not_ contain incrementals in that full dir, we default to storing the incrementals in `./incrementals` dir. 

So if the user is trying to mimic the old behavior for, the full proof method is to use the `incremental_location` parameter.

msbutler commented on v22.1/take-full-and-incremental-backups.md:
```diff
@@ -14,10 +14,62 @@ There are two main types of backups:
+    |—— 09-155340.13/
+      |—— Full backup files
+[...]
+|—— incrementals
```
is it worth writing out the full path to the incremental backup files, as you do in the second example? I.e., within `./incrementals`, you have:

```
|—— 2022
  |—— 02
    |—— 25-172907.21/
      |—— 20220325
        |—— 17921.23
          |—— incremental backup files
```

kathancox replied:
I can add this, if you feel it would be clearer

msbutler commented on v22.1/take-full-and-incremental-backups.md:
```diff
@@ -14,10 +14,62 @@ There are two main types of backups:
+By default, full backups are stored at the root of the collection's URI in a date-based path, and incremental backups are stored in the `/incrementals` directory. The following example shows a backup collection created using these default values, where all backups reside in one storage bucket:
+
+~~~
+Collection:
```
nit: sub Collection for CollectionURI

msbutler commented on v22.1/take-full-and-incremental-backups.md:
```diff
@@ -14,10 +14,62 @@ There are two main types of backups:
+
+[`SHOW BACKUPS IN {collectionURI}`](show-backup.html#view-a-list-of-the-available-full-backup-subdirectories) will display a list of the full backup subdirectories at the collection's URI.
+
+<a name="incremental-location-structure"></a> Alternately, the following directories also constitute a backup collection. There are multiple backups in two separate collection URIs. Each individual backup is a full backup and its related incremental backup(s). Despite using the [`incremental_location`](#incremental-backups-with-explicitly-specified-destinations) option to store the incremental ba ...
```
nit: in the first sentence, use 'separate URIs' instead of 'separate collection URIs' . A collection will always have only one collectionURI, but it may have multiple localityURis or explicit_incrementalURIs

msbutler replied:
also, instead of "cloud storage bucket", use "collectionURI"?

msbutler commented on v22.1/take-full-and-incremental-backups.md:
```diff
@@ -14,10 +14,62 @@ There are two main types of backups:
+<a name="incremental-location-structure"></a> Alternately, the following directories also constitute a backup collection. There are multiple backups in two separate collection URIs. Each individual backup is a full backup and its related incremental backup(s). Despite using the [`incremental_location`](#incremental-backups-with-explicitly-specified-destinations) option to store the incremental ba ...
+
+~~~
+Cloud storage bucket 1
```
maybe use CollectionURI instead of Cloud storage bucket ?

msbutler commented on v22.1/take-full-and-incremental-backups.md:
```diff
@@ -14,10 +14,62 @@ There are two main types of backups:
+~~~
+
+~~~
+Cloud storage bucket 2
```
maybe use "explicit_incrementalsURI" instead of "Cloud storage bucket 2"?

msbutler reviewed (commented):
no better way to treat covid than sleep, water, and pr reviews :D

msbutler commented on v22.1/take-full-and-incremental-backups.md:
```diff
@@ -92,45 +165,74 @@ Periodically run the [`BACKUP`][backup] command to take a full backup of your cl
+<span class="version-tag">New in v22.1:</span> This will add the incremental backup to the default `/incrementals` directory at the root of the backup collection's directory. With incremental backups in the `/incrementals` directory, you can apply different lifecycle/retention policies from cloud storage providers to the `/incrementals` directory as needed.
+
+{{site.data.alerts.callout_info}}
+In v21.2 and earlier, incremental backups were stored in the same directory as their full backup (i.e., `collectionURI/subdirectory`). If an incremental backup points to a subdirectory with incremental backups created in v21.2 and earlier, v22.1 will write the incremental backup to the v21.2 default location. To back up using the prior behavior, use the `incremental_location` option, as shown in  ...
```
nit: in the second sentence, add "command" , i.e. change to " If an incremental backup command points..."

per discussion in comments, perhaps clarify the last sentence to: "If you want to continue using the prior behavior on a backup that does not already contain incremental backups in the full backup subdirectory, use ..."

msbutler reviewed (approved):
@kathancox thanks for bearing with all my nits. Added two non-blocking comments

andf-crl reviewed (approved):
LGTM with just a few tiny optional nits. Great job with this!!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/13448#-:-N082ihC85CpWZxLIjzh:b1cycvi)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @kathancox, @livlobo, and @msbutler)

___
*[_includes/cockroachcloud/restore-examples.md, line 19 at r4](https://reviewable.io/reviews/cockroachdb/docs/13448#-N01QUvl21OuENp0cgj5:-N07uiyp6ny5r5_9Q9vO:b62y4c5) ([raw file](https://github.com/cockroachdb/docs/blob/e6946e39d160489500a68d45d5b6b2623d19e7c7/_includes/cockroachcloud/restore-examples.md#L19)):*
<details><summary><i>Previously, andf-crl wrote…</i></summary><blockquote>

nit: to restore a backup from

or

from which to restore a backup

or, from `SHOW BACKUPS`:

To view the backup paths in a given destination

x3 on this page
</blockquote></details>

nit-reversal: apologies, this can stand as is -- either way. I now know subdirectories usage better, and the original phrasing is appropriate as is. New is fine too. Up to you.
___
*[v21.1/restore.md, line 20 at r7](https://reviewable.io/reviews/cockroachdb/docs/13448#-N07vRTv5ymYV3SGZ5tI:-N07vRTv5ymYV3SGZ5tJ:b15n40x) ([raw file](https://github.com/cockroachdb/docs/blob/2b58c748f106251b0ba321d6f4855cd12890841d/v21.1/restore.md#L20)):*
> ```Markdown
> The [`BACKUP ... TO`](../v20.2/backup.html) and [`RESTORE ... FROM`](../v20.2/restore.html) syntax is **deprecated** as of v22.1 and will be removed in a future release.
> 
> We recommend using the `BACKUP ... INTO {collection}` syntax, which creates or adds to a [backup collection](take-full-and-incremental-backups.html#backup-collections) in your storage location. For restoring backups, we recommend using `RESTORE FROM {backup} IN {collection}` with `{backup}` being [`LATEST`](#restore-the-most-recent-backup) or a specific [subdirectory](#subdir-param).
> ```

nit: should both instances of `{collection}` be `{collectionURI}` ?
___
*[v21.2/restore.md, line 21 at r7](https://reviewable.io/reviews/cockroachdb/docs/13448#-N07vZJW2OzZCYhXk005:-N07vZJW2OzZCYhXk006:b15n40x) ([raw file](https://github.com/cockroachdb/docs/blob/2b58c748f106251b0ba321d6f4855cd12890841d/v21.2/restore.md#L21)):*
> ```Markdown
> The [`BACKUP ... TO`](../v20.2/backup.html) and [`RESTORE ... FROM`](../v20.2/restore.html) syntax is **deprecated** as of v22.1 and will be removed in a future release.
> 
> We recommend using the `BACKUP ... INTO {collection}` syntax, which creates or adds to a [backup collection](take-full-and-incremental-backups.html#backup-collections) in your storage location. For restoring backups, we recommend using `RESTORE FROM {backup} IN {collection}` with `{backup}` being [`LATEST`](#restore-the-most-recent-backup) or a specific [subdirectory](#subdir-param).
> ```

nit: should both instances of `{collection}` be `{collectionURI}` ?
___
*[v21.2/take-full-and-incremental-backups.md, line 20 at r7](https://reviewable.io/reviews/cockroachdb/docs/13448#-N07vbzMA-WzFYLM3C18:-N07vbzMA-WzFYLM3C19:b15n40x) ([raw file](https://github.com/cockroachdb/docs/blob/2b58c748f106251b0ba321d6f4855cd12890841d/v21.2/take-full-and-incremental-backups.md#L20)):*
> ```Markdown
> The [`BACKUP ... TO`](../v20.2/backup.html) and [`RESTORE ... FROM`](../v20.2/restore.html) syntax is **deprecated** as of v22.1 and will be removed in a future release.
> 
> We recommend using the `BACKUP ... INTO {collection}` syntax, which creates or adds to a [backup collection](take-full-and-incremental-backups.html#backup-collections) in your storage location. For restoring backups, we recommend using `RESTORE FROM {backup} IN {collection}` with `{backup}` being [`LATEST`](restore.html#restore-the-most-recent-backup) or a specific [subdirectory](restore.html#subdir-param).
> ```

nit: should both instances of `{collection}` be `{collectionURI}` ?
___
*[v21.2/take-full-and-incremental-backups.md, line 115 at r7](https://reviewable.io/reviews/cockroachdb/docs/13448#-N08-Zk38ucLagxiuYTO:-N08-Zk38ucLagxiuYTP:b-n1potw) ([raw file](https://github.com/cockroachdb/docs/blob/2b58c748f106251b0ba321d6f4855cd12890841d/v21.2/take-full-and-incremental-backups.md#L115)):*
> ```Markdown
> {{site.data.alerts.callout_info}}
> A full cluster restore can only be run on a target cluster that has **never** had user-created databases or tables.
> {{site.data.alerts.end}}
> ```

This section looks great, nice work! Only nit: the "a full cluster restore" alert should be indented under "to restore the latest ... full cluster". Not sure if that's possible yet for `{{site.data.alerts.callout_info}}`
___
*[v21.2/take-full-and-incremental-backups.md, line 124 at r7](https://reviewable.io/reviews/cockroachdb/docs/13448#-N08-zfvCfYLdHlx17aP:-N08-zfvCfYLdHlx17aQ:b2ls6my) ([raw file](https://github.com/cockroachdb/docs/blob/2b58c748f106251b0ba321d6f4855cd12890841d/v21.2/take-full-and-incremental-backups.md#L124)):*
> ```Markdown
>     ~~~
> 
> To view the available backup subdirectories, use [`SHOW BACKUPS`](show-backup.html).
> ```

nit: this should also be indented to be under "To restore ... specific subdirectory"
___
*[v22.1/manage-a-backup-schedule.md, line 68 at r4](https://reviewable.io/reviews/cockroachdb/docs/13448#-N01wWzDDxfg9QahTAh4:-N080Uyc6zRqsr5dJBnI:bd4zx5r) ([raw file](https://github.com/cockroachdb/docs/blob/e6946e39d160489500a68d45d5b6b2623d19e7c7/v22.1/manage-a-backup-schedule.md#L68)):*
<details><summary><i>Previously, andf-crl wrote…</i></summary><blockquote>

nit: same, should these be updated to match elsewhere:

- `SHOW BACKUPS IN collectionURI` statement ...
- `SHOW BACKUP FROM subdirectory IN collectionURI` statement ...

</blockquote></details>

I believe this nit still applies. Apologies if not!
___
*[v22.1/take-full-and-incremental-backups.md, line 25 at r4](https://reviewable.io/reviews/cockroachdb/docs/13448#-N023qOyCG8xsMOCBA1E:-N081vZ169lvZgIubeoR:bhm57iq) ([raw file](https://github.com/cockroachdb/docs/blob/e6946e39d160489500a68d45d5b6b2623d19e7c7/v22.1/take-full-and-incremental-backups.md#L25)):*
<details><summary><i>Previously, andf-crl wrote…</i></summary><blockquote>

nit: should we link URI ("collection URI "per michael) to BACKUP's params entry for `collectionURI`, since this is its first mention? Might need an anchor link?
</blockquote></details>

nit: I think agree with Michael to update [UIR] --> [collection URI]
___
*[v22.1/take-full-and-incremental-backups.md, line 144 at r7](https://reviewable.io/reviews/cockroachdb/docs/13448#-N082CRD4W4fjO827ETL:-N082CRD4W4fjO827ETM:b2kj2iz) ([raw file](https://github.com/cockroachdb/docs/blob/2b58c748f106251b0ba321d6f4855cd12890841d/v22.1/take-full-and-incremental-backups.md#L144)):*
> ```Markdown
>     ~~~
> 
> To view the available backup subdirectories, use [`SHOW BACKUPS`](show-backup.html).
> ```

same nits: indentation for the "a full cluster restore" alert and for the "to view the available backup subdirectories" sentence.


<!-- Sent from Reviewable.io -->
